ambry-api/src/main/java/com.github.ambry/config/ConnectionPoolConfig.java

```diff
@@ -60,6 +60,14 @@
   @Default("2")
   public final int connectionPoolMaxConnectionsPerPortSSL;
 
+  /**
+   * The maximum consecutive connect failures allowed before we clean up the available pool
+   */
+  @Config("connectionpool.max.connect.failures.to.clean.up.available.pool")
```

Comment:
I still feel this should be kept simple and we should clean up all of available pool on first failure. If a new connection initialization to a node fails, what are the chances of already established connections still being valid? What really is the argument for duplicating failure detector's logic here?